### PR TITLE
Fix package_libpng_1_6_7 to link explicitly against toolshed zlib

### PR DESCRIPTION
**Pull request by pjbriggs**
This PR updates `package_libpng_1_6_7` to explicitly set the `include` and `lib` directories for the version of the `zlib` libraries provided by its dependency `package_zlib_1_2_8`.

Without this patch, on Scientific Linux 6.2 the installation of `package_libpng_1_6_7` fails with an error of the form:

    contrib/tools/pngfix.o: In function `zlib_reset':
    /opt/apps/galaxy/tmp/tmp-toolshed-mtdvSxsyE/libpng-1.6.7/contrib/tools/pngfix.c:
    2175: undefined reference to `inflateReset2'
    collect2: ld returned 1 exit status
    make[1]: *** [pngfix] Error 1
    make: *** [all] Error 2

My patch uses the solution suggested in this StackOverflow answer: http://stackoverflow.com/a/26367243/579925

I don't know if there is a more canonical way to set the environment variables within `<action type="autoconf"/>`, so I have just swapped it for what I understand are the equivalent shell commands. Any comments or questions welcome - thanks for your consideration.

### CODE REVIEW

File: packages/package_libpng_1_6_7/tool_dependencies.xml [hunk 1]
```diff
@@ -12,7 +12,13 @@
...
           </repository>
         </action>
-        <action type="autoconf"/>
+	<!--
+	   Need to explicitly specify location of ZLIB files
+	   For background see
+	   http://stackoverflow.com/questions/19738464/pngfix-c2151-undefined-reference-to-inflatereset2
+	  -->
+	<action type="shell_command">LDFLAGS=-L$ZLIB_ROOT_PATH/lib CPPFLAGS=-I$ZLIB_ROOT_PATH/include ./configure --prefix=$INSTALL_DIR &amp;&amp;
+	  make &amp;&amp; make install</action>
```
Comment (bgruening) [inline packages/package_libpng_1_6_7/tool_dependencies.xml]: This can be replaced by the `make_install` action

Comment (pjbriggs) [inline packages/package_libpng_1_6_7/tool_dependencies.xml]: @bgruening I've split out the `make_install` action, however the `configure` command is still in a `shell_command` action. Is there a `configure` action type? I can't see it in the docs at https://wiki.galaxyproject.org/ToolDependenciesTagSets

**Comment by pjbriggs** [comment]
Looks to me like this problem has been fixed (and more elegantly) by PR #578, which has just been merged.

If that update could be pushed to the TS then I think that would solve the problems that this was intended to fix.

**Comment by yhoogstrate** [comment]
@pjbriggs The tool sheds are now updated. Could you report whether this solves is also for your case?

**Comment by pjbriggs** [comment]
@yhoogstrate Thanks I think this has fixed it for me.

**Pull request closed without merging**